terraform/environments/electronic-monitoring-data/glue-job/parquet_to_csv.py

```diff
@@ -0,0 +1,86 @@
+import sys
+from awsglue.transforms import *
+from awsglue.utils import getResolvedOptions
+from pyspark.context import SparkContext
+from awsglue.context import GlueContext
+from awsglue.job import Job
+import boto3
+from logging import getLogger
+
+logger = getLogger(__name__)
+
+# Set up Glue context and job
+args = getResolvedOptions(sys.argv, ["JOB_NAME", "source_bucket", "destination_bucket"])
+sc = SparkContext()
+glueContext = GlueContext(sc)
+spark = glueContext.spark_session
+job = Job(glueContext)
+job.init(args["JOB_NAME"], args)
+s3 = boto3.client("s3")
+
+# Source and destination buckets
+source_bucket = args["source_bucket"]
+destination_bucket = args["destination_bucket"]
+source_path = f"s3://{source_bucket}/"
+destination_path = f"s3://{destination_bucket}/"
+
+logger.info(f"Reading from {source_bucket}, writing to {destination_bucket}.")
+
+
+def get_tables_from_s3_path(s3_client, bucket_name="dms-em-rds-output"):
+    # Pagination
+    paginator = s3_client.get_paginator("list_objects_v2")
+    response_iterator = paginator.paginate(Bucket=bucket_name)
+
+    for page in response_iterator:
+        keys = [obj["Key"] for obj in page.get("Contents", [])]
+    filered_keys = [key for key in keys if ".parquet" in key]
+
+    database_table_pairs = list(
+        set([(key.split("/")[0], key.split("/")[2]) for key in filered_keys])
+    )
+
+    result = {}
+
+    for key, value in database_table_pairs:
+        if key in result:
+            result[key].append(value)
+        else:
+            result[key] = [value]
+
+    # Convert the lists to sets to remove duplicate table names
+    for key in result:
+        result[key] = list(set(result[key]))
+
+    return result
```

Comment:
i'd probably drop `result` into the logs just for a sanity check if the glue job breaks when iterating through